### PR TITLE
solution

### PR DESCRIPTION
**Pull request by vmiskiv1**

### CODE REVIEW

**Review by AndreyMalyshev147 — APPROVED** [review]
Great work))🚀

File: src/averageAges.js [hunk 1]
```diff
@@ -15,11 +15,16 @@
...
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const malesBornInCentury = people.filter(person => person.sex === 'm'
+  && Math.ceil(person.died / 100) === century);
+
+  const allMales = people.filter(person => person.sex === 'm');
+
+  const males = century ? malesBornInCentury : allMales;
```
Comment (aliceblunt8) [inline src/averageAges.js:23]: first you can find all men and then if century is passed you can use the same array to filter males born in specific century
now you filter people array each time

File: src/averageAges.js [hunk 2]
```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
...
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const femalesWithChildren = people.filter(person =>
+    person.sex === 'f' && people.some(child => child.mother === person.name));
+
+  const allFemales = people.filter(person =>
+    person.sex === 'f');
+
+  const females = withChildren ? femalesWithChildren : allFemales;
```
Comment (aliceblunt8) [inline src/averageAges.js:51]: same here, first find all women, then filter only those with children

File: src/averageAges.js [hunk 3]
```diff
@@ -55,7 +69,24 @@ function calculateWomenAverageAge(people, withChildren) {
...
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const femaleWithSon = person =>
+    people.some(female => person.mother === female.name)
+    && person.sex === 'm';
+
+  const femaleWithChild = person =>
+    people.some(female => person.mother === female.name);
+
+  const isOnlyWithSon = people.filter(
```
Comment (aliceblunt8) [inline src/averageAges.js:79]: isOnlyWithSon is name for a boolean value, not for array list

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**